src/runtime/recipe/annotation.ts

```diff
@@ -31,24 +32,26 @@ export class Annotation implements Comparable<Annotation> {
     return 0;
   }
 
-  toString(): string {
-    const result: string[] = [];
+  toManifestString(builder?: ManifestStringBuilder): string {
+    builder = builder || new ManifestStringBuilder();
     let paramStr = '';
     if (Object.keys(this.params).length > 0) {
       paramStr = `(${Object.keys(this.params).map(name => `${name}: ${this.params[name]}`).join(', ')})`;
     }
-    result.push(`annotation ${this.name}${paramStr}`);
-    if (this.targets.length > 0) {
-      result.push(`  targets: [${this.targets.join(', ')}]`);
-    }
-    result.push(`  retention: ${this.retention}`);
-    if (this.allowMultiple) {
-      result.push(`  allowMultiple: ${this.allowMultiple}`);
-    }
-    if (this.doc) {
-      result.push(`  doc: '${this.doc}'`);
-    }
-    return result.filter(s => s !== '').join('\n');
+    builder.push(`annotation ${this.name}${paramStr}`);
```

Comment:
optional - can we do full chaining maybe, like:
`builder.push(...).withIndent(...)....toString();`
basically just `push` needs to return `this`.

Reply:
Sadly, no. `withIndent()` currently returns the new `indentedBuilder`, which means chaining a `toString()` call on the end will not return the entire string.